compiler/resolution/functionResolution.cpp

```diff
@@ -7314,13 +7314,26 @@ pruneResolvedTree() {
   removeCompilerWarnings();
 }
 
+static void clearDefaultInitFns(FnSymbol* unusedFn) {
+  // Before removing an unused function, check if it is the default initializer
+  // for any type.  If it is, set the defaultInitializer field for the type
+  // to NULL so the removed function doesn't leave behind a garbage pointer.
+  forv_Vec(TypeSymbol, ts, gTypeSymbols) {
```

Comment:
This loop might have performance impacts.
If we assume that the default initializer always returns an object of the containing AggregateType, you can use the return type of the function to find its parent and avoid the search.

Some time in the future, the defaultInitializer will be converted into a method, in which case, the type of its implicit "this" argument will be used instead.  It might be helpful to add a comment to that effect.